packages/shared-ux/prompt/errors/src/not_found.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useState, useCallback } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiEmptyPrompt,
+  EuiEmptyPromptProps,
+  EuiImage,
+  useEuiTheme,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+const NOT_FOUND_TITLE = i18n.translate('sharedUXPackages.prompt.errors.notFound.title', {
+  defaultMessage: 'Page not found',
+});
+
+const NOT_FOUND_BODY = i18n.translate('sharedUXPackages.prompt.errors.notFound.body', {
+  defaultMessage:
+    "Sorry, the page you're looking for can't be found. It might have been removed or renamed, or maybe it never existed at all.",
+});
+
+const NOT_FOUND_GO_BACK = i18n.translate('sharedUXPackages.prompt.errors.notFound.goBacklabel', {
+  defaultMessage: 'Go back',
+});
+
+interface NotFoundProps {
+  /** Array of buttons, links and other actions to show at the bottom of the `EuiEmptyPrompt`. Defaults to a "Back" button. */
+  actions?: EuiEmptyPromptProps['actions'];
+}
+
+/**
+ * Predefined `EuiEmptyPrompt` for 404 pages.
+ */
+export function NotFound({ actions }: NotFoundProps) {
+  const { colorMode } = useEuiTheme();
+  const [errorImage, setErrorImage] = useState<string>();
+  const goBack = useCallback(() => history.back(), []);
+
+  useEffect(() => {
+    const loadImage = async () => {
+      const { default: img } = await import(
+        `./assets/404_astronaut_${colorMode.toLowerCase()}.png`
+      );
+      setErrorImage(img);
+    };
+    loadImage();
+  }, [colorMode]);
+
+  const icon = errorImage ? <EuiImage src={errorImage} alt="" /> : null;
+
+  return (
+    <EuiEmptyPrompt
+      color="subdued"
+      titleSize="m"
+      icon={icon}
+      title={<h2>{NOT_FOUND_TITLE}</h2>}
+      body={NOT_FOUND_BODY}
+      actions={
+        actions ?? [
+          <EuiButtonEmpty iconType="arrowLeft" flush="both" onClick={goBack}>
```

Comment:
nit: declare this as a const above within the functional component.  Otherwise you risk recreating the element on each render, (referential difference vs. shallow comparison).  It also makes the `goBack` const cleaner.